File: sdk/golang/pkg/generate/margin/marginpublic/api_margin_public_test.go
package marginpublic

import (
	"encoding/json"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/types"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestMarginPublicIndexPriceRespModel(t *testing.T) {
	// IndexPrice
	// Index Price

	data := "{\"id\":\"5c24c5da03aa673885cd67a0\",\"type\":\"message\",\"topic\":\"/indicator/index:USDT-BTC\",\"subject\":\"tick\",\"data\":{\"symbol\":\"USDT-BTC\",\"granularity\":5000,\"timestamp\":1551770400000,\"value\":0.0001092}}"

	commonResp := &types.WsMessage{}
	err := json.Unmarshal([]byte(data), commonResp)
	assert.Nil(t, err)
	assert.NotNil(t, commonResp.RawData)
	obj := &IndexPriceEvent{}
	err = json.Unmarshal(commonResp.RawData, obj)
	assert.Nil(t, err)
}

func TestMarginPublicMarkPriceRespModel(t *testing.T) {
	// MarkPrice
	// Mark Price

	data := "{\"id\":\"5c24c5da03aa673885cd67aa\",\"type\":\"message\",\"topic\":\"/indicator/markPrice:USDT-BTC\",\"subject\":\"tick\",\"data\":{\"symbol\":\"USDT-BTC\",\"granularity\":5000,\"timestamp\":1551770400000,\"value\":0.0001093}}"

	commonResp := &types.WsMessage{}
	err := json.Unmarshal([]byte(data), commonResp)
	assert.Nil(t, err)
	assert.NotNil(t, commonResp.RawData)
	obj := &MarkPriceEvent{}
	err = json.Unmarshal(commonResp.RawData, obj)
	assert.Nil(t, err)
}

File: sdk/python/kucoin_universal_sdk/generate/margin/margin_public/api_margin_public_test.py
import unittest
from .model_index_price_event import IndexPriceEvent
from .model_mark_price_event import MarkPriceEvent
from typing import List
from kucoin_universal_sdk.model.common import WsMessage


class MarginPublicAPITest(unittest.TestCase):

    def test_index_price_resp_model(self):
        """
        index_price
        Index Price
        /indexPrice/indicator/index:_symbol_,_symbol_
        """
        data = "{\"id\":\"5c24c5da03aa673885cd67a0\",\"type\":\"message\",\"topic\":\"/indicator/index:USDT-BTC\",\"subject\":\"tick\",\"data\":{\"symbol\":\"USDT-BTC\",\"granularity\":5000,\"timestamp\":1551770400000,\"value\":0.0001092}}"
        common_response = WsMessage.from_json(data)
        resp = IndexPriceEvent.from_dict(common_response.raw_data)

    def test_mark_price_resp_model(self):
        """
        mark_price
        Mark Price
        /markPrice/indicator/markPrice:_symbol_,_symbol_
        """
        data = "{\"id\":\"5c24c5da03aa673885cd67aa\",\"type\":\"message\",\"topic\":\"/indicator/markPrice:USDT-BTC\",\"subject\":\"tick\",\"data\":{\"symbol\":\"USDT-BTC\",\"granularity\":5000,\"timestamp\":1551770400000,\"value\":0.0001093}}"
        common_response = WsMessage.from_json(data)
        resp = MarkPriceEvent.from_dict(common_response.raw_data)
